### ISSUE TITLE
[infra] infra/README.md beginner guide (macOS + Linux) + Disaster Recovery section

### ISSUE THREAD

**Issue by denisvmedia** [denisvmedia/inventario#1863]
**Updated 2026-05-24** per ArgoCD pivot ([#1852](https://github.com/denisvmedia/inventario/issues/1852#issuecomment-4528693792)). Bot operations walkthrough replaced by ArgoCD UI walkthrough; magic-comment workflow replaced by `preview` label workflow.

## Scope

User-facing README at `infra/README.md` that a brand-new contributor (who's never touched ArgoCD/helm/k8s) can follow start-to-finish.

### Sections

1. **What this is** — one paragraph: what `make bootstrap` gives you and why. ASCII diagram of VM → vcluster → ArgoCD → preview namespaces.
2. **Prerequisites** — exact commands to install on laptop:
   - `sops, age, helm, kubectl, ssh, tailscale` CLI
   - macOS (Homebrew) + Linux (apt for Debian/Ubuntu, with notes for arch/fedora)
3. **First-time setup**:
   - Generate age key (`age-keygen -o ~/.config/sops/age/keys.txt`)
   - **Back up age key** (1Password / Bitwarden / etc.) — **explicit warning, loss = total DR failure**
   - Obtain Tailscale OAuth client (link to TS admin console step-by-step)
   - Create GitHub App (step-by-step per #1854 README block — Permissions: Contents/Metadata/Pull requests Read, webhook disabled, install on this repo, save App ID + Installation ID, generate private key)
   - `sops infra/vm/secrets/secrets.enc.yaml` to populate all secrets
   - Provision a VM (links to Hetzner / DigitalOcean / etc. — generic)
   - `make bootstrap VM=user@vm-ip`
   - Verify: `tailscale status` shows `inv-vcl01`; `kubectl -n argocd get pods` all Ready; ArgoCD UI loads via port-forward
4. **Day-to-day**:
   - **Trigger a preview**: add label `preview` to an open PR. Within ~60s an Application `inv-vcl01-prNNNN` appears in ArgoCD UI. Within ~5 minutes the app is reachable at `https://inv-vcl01-prNNNN.<tailnet>.ts.net/`.
   - **Tear down a preview**: remove label `preview` from PR, OR close/merge the PR. ArgoCD prunes within one poll cycle.
   - **Inspect a preview**:
     ```bash
     kubectl -n argocd port-forward svc/argocd-server 8080:80
     # open http://localhost:8080
     # admin password:
     kubectl -n argocd get secret argocd-initial-admin-secret \
       -o jsonpath='{.data.password}' | base64 -d
     ```
   - **master deployment**: auto-syncs to master HEAD on every push (Application `inv-vcl01-master`); reachable at `https://inv-vcl01-master.<tailnet>.ts.net/`.
   - **Make targets**: `make status` (cluster overview), `make logs SVC=argocd-server`, `make shell` (SSH to VM), etc.
5. **Troubleshooting**:
   - **Preview not appearing** → check ApplicationSet status (`kubectl -n argocd describe appset inventario-pr-previews`), confirm label is exactly `preview` (case-sensitive), check `argocd-applicationset-controller` logs for GH API errors.
   - **Application stuck Syncing** → ArgoCD UI app status; common: GHCR build for `sha-<7>` not done yet (wait ~5 min) or chart parse error.
   - **GHCR pull errors** → not expected (image is public per [#1867 spike](https://github.com/denisvmedia/inventario/issues/1867#issuecomment-4528467749)); check `kubectl describe pod` for actual error.
   - **TS-operator stuck** → `kubectl -n tailscale logs deploy/tailscale-operator`; common: OAuth client expired or scope insufficient.
   - **Ingress :443 not reachable** → `tailscale status` from a peer, confirm operator-provisioned tailnet node appeared.
   - **ArgoCD UI 401** → admin secret rotated; reset via `argocd admin initial-password` from inside argocd-server pod.
   - **GitHub App API rate limit** → `kubectl -n argocd logs deploy/argocd-applicationset-controller | grep -i ratelimit`; bump `requeueAfterSeconds` if needed.
6. **Disaster Recovery** — "your VM is gone" runbook:
   1. Spin up new Ubuntu 26.04 VM
   2. `ssh-copy-id`
   3. `make recover VM=user@new-vm-ip`
   4. (Phase 2) `make restore-longevity` once #1865 lands
   5. Verify (`make status`, `kubectl -n argocd get applications`)
   6. Why you don't need to notify anyone (same TS hostnames, same admin creds, same ArgoCD App)
7. **GitHub App key rotation** — see #1854 procedure; cross-link.
8. **Phase 2 future** — pointer to Velero (#1865) + Resource quotas (#1866).

**Goal**: copy-paste runnable. No `...` or "fill in the blanks" without explicit `<MARKER>`.

## Acceptance

- A fresh contributor, given access to the repo + the age key + the GitHub App credentials, can `make bootstrap` and have a working preview env within 30 minutes without asking anyone questions.
- The DR section has been walked through end-to-end on a throwaway VM (destroy → recover → verify identity preserved + ArgoCD reconnects + Applications sync) before this issue closes.

## Dependencies

- #1853, #1854, #1855, #1856, #1858 — all must land before README can be written end-to-end.

## Non-goals

- API docs for Inventario itself (those live elsewhere).
- Multi-VM deployment (Phase 2+).
- ArgoCD UI customization (theme, plugins, SSO).
- Public exposure of ArgoCD UI (tailnet-only access via port-forward is the design).

**Comment by denisvmedia** [comment]
README addition request — surfaced while fresh-deploy-testing #1881.

Add under "Troubleshooting" (between "TS-operator stuck" and "Ingress :443 not reachable") or in a new "Tailscale tidying" subsection of the Disaster Recovery runbook:

### Stale Tailscale device records after snapshot rollback / cluster wipe

In normal operation, ephemeral PR-preview proxies self-clean (the `ephemeral-pr` ProxyClass passes `TS_EXTRA_ARGS=--ephemeral` to the proxy container). On a snapshot rollback, a force-deleted namespace, or any path that doesn't let `tailscaled` cleanly shut down, you may see stale offline devices in the TS admin holding the original hostname:

```
inv-vcl01-pr1881           offline, last seen 31m ago    100.83.x.x
inv-vcl01-pr1881-1         tagged-devices                 100.85.x.x  <- new proxy
```

The operator can't reclaim the original hostname while the offline record exists, so it appends `-1` / `-2` suffixes.

**Operator itself has the same property**: the operator pod stores its identity in a K8s Secret (`OPERATOR_SECRET=operator` in ns `tailscale`). When the cluster is wiped, that Secret is lost → operator re-registers as a new tailnet device. After multiple bootstrap re-runs you may see `tailscale-operator`, `tailscale-operator-1`, `tailscale-operator-2`, ...

Neither the chart nor the operator code currently expose an `Ephemeral` capability for the operator's own device (verified against `cmd/k8s-operator/operator.go` HEAD on 2026-05-25 — `caps.Devices.Create` sets `Reusable`, `Preauthorized`, `Tags` but not `Ephemeral`).

**Cleanup**: log into [admin.tailscale.com → Machines](https://login.tailscale.com/admin/machines), filter by `tag:k8s` (for proxies) or `tag:k8s-operator` (for the operator), remove the offline rows. Devices with default key expiry would otherwise persist for 180 days.

Snapshot rollback and full cluster wipes are dev/recovery flows, not production — in steady state nothing accumulates.
